indexing/src/test/java/com/google/enterprise/cloudsearch/sdk/indexing/template/RepositoryDocTest.java

```diff
@@ -96,6 +100,38 @@ public void testOnlyItem() throws IOException, InterruptedException {
     assertEquals("id1", doc.getItem().getName());
   }
 
+  @Test
+  public void testCallback() throws IOException, InterruptedException {
+    Item item = new Item().setName("id1").setAcl(getCustomerAcl());
+    final HashMap callbackResult = new HashMap();
+    RepositoryDoc doc = new RepositoryDoc.Builder()
+            .setItem(item)
+            .setCallback(new FutureCallback<GenericJson>() {
+              @Override
+              public void onSuccess(@Nullable GenericJson result) {
```

Comment:
onSuccess and onFailure are behaving the same, which makes the test less clear. a better way for verifying this is to set a mocked FutureCallback and verify that onSuccess() is called eventually.